flink-connectors/flink-connector-elasticsearch2/src/test/java/org/apache/flink/streaming/connectors/elasticsearch2/ElasticsearchSinkITCase.java

```diff
@@ -17,217 +17,51 @@
-
-	DataStreamSource<Tuple2<Integer, String>> source = env.addSource(new TestSourceFunction());
-
-	Map<String, String> config = new HashMap<>();
-	// This instructs the sink to emit after every element, otherwise they would be buffered
-	config.put(ElasticsearchSink.CONFIG_KEY_BULK_FLUSH_MAX_ACTIONS, "1");
-	config.put("cluster.name", "my-transport-client-cluster");
-
-	source.addSink(new ElasticsearchSink<>(config, new ArrayList<InetSocketAddress>(), new TestElasticsearchSinkFunction()));
-
-	env.execute("Elasticsearch TransportClient Test");
-
-	// verify the results
-	Client client = node.client();
-	for (int i = 0; i < NUM_ELEMENTS; i++) {
-	 GetResponse response = client.get(new GetRequest("my-index",
-		 "my-type", Integer.toString(i))).actionGet();
-	 Assert.assertEquals("message #" + i, response.getSource().get("data"));
+	@Test
+	public void testEmptyTransportClient() throws Exception {
+		runEmptyTransportClientTest();
 	}
 
-	node.close();
- }
-
-	@Test(expected = JobExecutionException.class)
+	@Test
 	public void testTransportClientFails() throws Exception{
-		// this checks whether the TransportClient fails early when there is no cluster to
-		// connect to. There isn't a similar test for the Node Client version since that
-		// one will block and wait for a cluster to come online
-
-		final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
-
-		DataStreamSource<Tuple2<Integer, String>> source = env.addSource(new TestSourceFunction());
-
-		Map<String, String> config = new HashMap<>();
-		// This instructs the sink to emit after every element, otherwise they would be buffered
-		config.put(ElasticsearchSink.CONFIG_KEY_BULK_FLUSH_MAX_ACTIONS, "1");
-		config.put("cluster.name", "my-node-client-cluster");
-
-		List<InetSocketAddress> transports = new ArrayList<>();
-		transports.add(new InetSocketAddress(InetAddress.getByName("127.0.0.1"), 9300));
-
-		source.addSink(new ElasticsearchSink<>(config, transports, new TestElasticsearchSinkFunction()));
-
-		env.execute("Elasticsearch Node Client Test");
+		runTransportClientFailsTest();
 	}
 
-	private static class TestSourceFunction implements SourceFunction<Tuple2<Integer, String>> {
-		private static final long serialVersionUID = 1L;
-
-		private volatile boolean running = true;
-
-		@Override
-		public void run(SourceContext<Tuple2<Integer, String>> ctx) throws Exception {
-			for (int i = 0; i < NUM_ELEMENTS && running; i++) {
-				ctx.collect(Tuple2.of(i, "message #" + i));
-			}
-		}
-
-		@Override
-		public void cancel() {
-			running = false;
-		}
+	@Override
+	protected <T> ElasticsearchSinkBase<T> createElasticsearchSink(Map<String, String> userConfig,
+																List<InetSocketAddress> transportAddresses,
+																ElasticsearchSinkFunction<T> elasticsearchSinkFunction) {
+		return new ElasticsearchSink<>(userConfig, transportAddresses, elasticsearchSinkFunction);
```

Comment:
Same as above comment: there don't seem to be trailing spaces.
If you meant empty lines, there are actually method parameters on the far right.